### PR TITLE
fix shenyu-plugin-apache-dubbo : set register to false.

### PR DESCRIPTION
**Pull request by yudayday**
[ISSUE #5242] set register to false.

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java [hunk 1]
```diff
@@ -230,11 +230,15 @@ private ReferenceConfig<GenericService> buildReference(final MetaData metaData,
             Optional.ofNullable(dubboParam.getSent()).ifPresent(reference::setSent);
         }
         if (StringUtils.isNotBlank(namespace)) {
-            if (!registryConfig.getAddress().contains(Constants.NAMESPACE)) {
-                reference.setRegistry(new RegistryConfig(registryConfig.getAddress() + "?" + Constants.NAMESPACE + "=" + namespace));
+            RegistryConfig registryConfig = new RegistryConfig();
+            registryConfig.setRegister(false);
+            if (!this.registryConfig.getAddress().contains(Constants.NAMESPACE)) {
+                registryConfig.setAddress(this.registryConfig.getAddress() + "?" + Constants.NAMESPACE + "=" + namespace);
```
Comment (moremind) [inline shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java:236]: why modify this?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java (hunk 1 above)
Comment (yudayday) [reply shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java:236]: see ISSUE https://github.com/apache/shenyu/issues/5242

Comment (moremind) [reply shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java:236]: why setRegister in configuration of dubbo application cache?

Comment (moremind) [reply shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java:236]: `org.apache.shenyu.plugin.apache.dubbo.cache.ApacheDubboConfigCache#init`

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java (hunk 1 above)
Comment (yudayday) [reply shenyu-plugin/shenyu-plugin-proxy/shenyu-plugin-rpc/shenyu-plugin-dubbo/shenyu-plugin-apache-dubbo/src/main/java/org/apache/shenyu/plugin/apache/dubbo/cache/ApacheDubboConfigCache.java:236]: 是的，在这个ApacheDubboConfigCache#init方法内，register是设置成false了，但是在ApacheDubboConfigCache#buildReference中，在namespace不为blank的情况下，没有设置成false
image

**Review by moremind — APPROVED** [review]

**Comment by dragon-zhang** [comment]
@yudayday good job, please add wechat 18179469818

**Pull request merged**